testing/server.go

```diff
@@ -106,12 +121,7 @@ func (s *Server) UploadCharm(c *gc.C, ch charm.Charm, id *charm.Reference, promu
 	}
 
 	// Upload the charm.
```

Comment:
The docs for this function should probably be clearer about how the parameters are used.
For example, it should mention that the revision must be specified and whether the
promulgated flag causes the base entity to be promulgated. (sorry, I can't comment
on the function declaration itself)